ddtelemetry/src/worker/mod.rs

```diff
@@ -167,7 +170,11 @@ impl TelemetryWorker {
 
             match self.dispatch_metrics_logs_action(action).await {
                 ControlFlow::Continue(()) => {}
-                ControlFlow::Break(()) => break,
+                ControlFlow::Break(()) => {
+                    if !self.config.restartable {
```

Comment:
It would be nice to have some test that shows that the restartable feature works as expected.

Reply:
It would, however there are currently no tests for the overall functionality of the worker lifecycle, which I could extend.
This PR is not the place to add a full-blown testsuite.

Reply:
The joys of _shared_ ownership...